File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.java
import java.util.HashSet;
class Solution {
	public int lengthOfLongestSubstring(String s) {
		HashSet<Character> set = new HashSet<Character>();
		int head = 0; int tail = 0; int count = 0;
		while (head < s.length()) {
			while(set.contains(s.charAt(head))) {
				set.remove(s.charAt(tail++));
			}
			set.add(s.charAt(head++));
			count = Math.max(head-tail, count);
		}
		return count;
	}
}

File: LeetCode Solutions/3. Longest Substring Without Repeating Characters/solution3.py
class Solution:
	def lengthOfLongestSubstring(self, s: str) -> int:
		head = 0; tail = 0; lenSet = set(); count = 0; sol = 0
		while head < len(s):
			while s[head] in lenSet:
				lenSet.remove(s[tail])
				tail += 1
			lenSet.add(s[head])
			head += 1
			count = head-tail
			if (count > sol):
				sol = count
		return sol